### PR TITLE
[NIL] New NIL namespace and indicator document type

### PR DESCRIPTION
**Pull request by hugosykes**
New nationalindicatorlibrary namespace created for use by NIL team. Created new document type with required fields to facilitate migration. Render pipeline complete between new document in CMS and rendering on local site.

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/national-indicator-library.yaml [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      - ''
+      - ''
+      - ''
+      - ''
+      - ''
+      - ''
+      - ''
+      resourcebundle:id: nationalindicatorlibrary.headers
+      resourcebundle:keys:
+      - publishedBy
```
Comment (jacsamell) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/national-indicator-library.yaml]: can you prefix all of these with `headers.`, we have a few different resorce files so it's good to be able to differentiate between then

File: repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/publication-system/ci-hub.yaml [hunk 2]
```diff
@@ -3,33 +3,101 @@
   /ccg-outcomes:
     /ccg-outcomes[1]:
       /publicationsystem:content:
-        hippostd:content: ''
```
Comment (jacsamell) [inline repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/publication-system/ci-hub.yaml]: auto-exported, please revert

File: repository-data/development/src/main/resources/hcm-config/main.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 definitions:
   config:
     /hippo:configuration/hippo:modules/autoexport/hippo:moduleconfig:
-      autoexport:enabled: false
```
Comment (jacsamell) [inline repository-data/development/src/main/resources/hcm-config/main.yaml]: Remember to turn auto export off before pushing 🙂 
Revert this.

File: repository-data/development/src/main/resources/hcm-content/content/documents/corporate-website/national-indicator-library.yaml [hunk 4]
```diff
@@ -0,0 +1,260 @@
+---
+/content/documents/corporate-website/national-indicator-library:
+  /sample-indicator:
```
Comment (jacsamell) [inline repository-data/development/src/main/resources/hcm-content/content/documents/corporate-website/national-indicator-library.yaml]: It's up to you NIL guys but we have been putting each folder/doc in a separate yaml file, but you guys don't have to follow us on that if you don't want 🙂

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+
+
+    <section class="document-content">
+    </section>
+
+
+  <footer class="footer">
+    <div class="footer__inner">
+      FOOTER
```
Comment (jacsamell) [inline repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/indicator.ftl]: Probably just want to use the footer from base page to be consistent with the rest of the site but you can sort that later down the line

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/macro/structured-text.ftl [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<#ftl output_format="HTML">
+<#include "../../include/imports.ftl">
+
+<#macro structuredText item uipath>
```
Comment (jacsamell) [inline repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/macro/structured-text.ftl]: If this is duplicate of publication system one then move it to common so we don't have duplicates

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/publication.ftl [hunk 7]
```diff
@@ -4,7 +4,7 @@
 <#assign dateFormat="dd/MM/yyyy"/>
 <#assign formatFileSize="uk.nhs.digital.ps.directives.FileSizeFormatterDirective"?new() />
 <#assign formatRestrictableDate="uk.nhs.digital.ps.directives.RestrictableDateFormatterDirective"?new() />
-<#assign formatCoverageDates="uk.nhs.digital.ps.directives.CoverageDatesFormatterDirective"?new() >
```
Comment (jacsamell) [inline repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/publication.ftl]: ?

File: site/src/main/java/uk/nhs/digital/nil/components/IndicatorComponent.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        final HstRequestContext ctx = request.getRequestContext();
+        Indicator indicator = getIndicator(ctx);
+
+        request.setAttribute("indicator", indicator);
+    }
+
+    private Indicator getIndicator(HstRequestContext ctx) throws HstComponentException {
+        HippoBean content = ctx.getContentBean();
+
+        if (content.getClass().equals(Indicator.class)) {
```
Comment (jacsamell) [inline site/src/main/java/uk/nhs/digital/nil/components/IndicatorComponent.java]: `if (content instanceof Indicator)`

File: styleguide/css/style.css [hunk 9]
```diff
@@ -539,9 +539,11 @@ a {
 /* BREADCRUMBS */
 /* ########################################################## */
 div.breadcrumb {
-  border-bottom: 1px solid #768692;
```
Comment (jacsamell) [inline styleguide/css/style.css]: Are these breadcrumb changes intentional?

File: repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/macro/structured-text.ftl (hunk 6 above)
Comment (hugosykes) [reply repository-data/webfiles/src/main/resources/site/freemarker/nationalindicatorlibrary/macro/structured-text.ftl]: Not necessary in nil folder - now deleted

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/national-indicator-library.yaml (hunk 1 above)
Comment (hugosykes) [reply repository-data/application/src/main/resources/hcm-content/content/documents/administration/national-indicator-library.yaml]: sorted

File: repository-data/application/src/main/resources/hcm-config/hst/configurations.yaml [hunk 10]
```diff
@@ -4,6 +4,10 @@ definitions:
     /hst:hst/hst:configurations:
       /common:
         jcr:primaryType: hst:configuration
+      /nationalindicatorlibrary:
+        hst:inheritsfrom:
+        - ../common
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-config/hst/configurations.yaml]: I would suggest you don't extend common

File: repository-data/application/src/main/resources/hcm-config/hst/configurations/nationalindicatorlibrary/workspace/channel.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+---
+definitions:
+  config:
+    /hst:hst/hst:configurations/nationalindicatorlibrary/hst:workspace/hst:channel:
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-config/hst/configurations/nationalindicatorlibrary/workspace/channel.yaml]: there should be no channel

File: repository-data/application/src/main/resources/hcm-content/hst/configurations/nationalindicatorlibrary/channelinfo.yaml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+---
+/hst:hst/hst:configurations/nationalindicatorlibrary/hst:workspace/hst:channel/hst:channelinfo:
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-content/hst/configurations/nationalindicatorlibrary/channelinfo.yaml]: again, we should have just one channel for "the website"

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/publication.ftl (hunk 7 above)
Comment (hugosykes) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/publication.ftl]: missing '/' at the end

Comment (jacsamell) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/publication.ftl]: 👍

**Pull request merged**